app/client/src/utils/index.ts

```diff
@@ -14,3 +14,9 @@ export function debounce<T extends Callback>(cb: T, timeout: number): Callback {
         }, timeout);
     };
 }
+// source link for Regex https://stackoverflow.com/questions/3809401/what-is-a-good-regular-expression-to-match-a-url
+export function isURL(str: string) {
+    const pattern = new RegExp(/[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)?/gi); 
+    return !!pattern.test(str);
+}
```

Comment:
Looking at this again, it may be more prudent to try to create a `URL()` and if an error is not thrown, then we know that it's a URL. https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

That's probably not the best way to about using the API, but at the same time, whatever they're using to validate a URL should be much more robust than anything we'd make.

Reply:
resolved in other thread